pandas/core/arrays/string_arrow.py

```diff
@@ -0,0 +1,486 @@
+from __future__ import annotations
+
+from collections import abc
+from distutils.version import LooseVersion
+from typing import Any, Optional, Sequence, Type, Union
+
+import numpy as np
+
+from pandas._libs import lib, missing as libmissing
+from pandas._typing import ArrayLike
+
+from pandas.core.dtypes.base import ExtensionDtype
+from pandas.core.dtypes.dtypes import register_extension_dtype
+
+import pandas as pd
+from pandas.api.types import (
+    is_array_like,
+    is_bool_dtype,
+    is_integer,
+    is_integer_dtype,
+    is_scalar,
+)
+from pandas.core.arrays.base import ExtensionArray
+from pandas.core.indexers import check_array_indexer
+
+try:
+    import pyarrow as pa
+except ImportError:
+    pa = None
+else:
+    try:
+        import pyarrow.compute as pc
+    except ImportError:
+        pass
+
+
+def _as_pandas_scalar(arrow_scalar: pa.Scalar) -> Optional[str]:
```

Comment:
also how is the return type Optional?

Reply:
Not sure how to correctly type the return type here. In the case it is NA, it isn't a string. `Optional` sounded fitting but `self.na_value` is probably not matching with what `mypy` expects.

Reply:
yes. not so easy. but I think we definitely want to remove the Optional.

Reply:
I suppose strictly speaking it is `Union[str, type(pd.NA)]`, would be nice to have a shortcut for the general "Scalar-type or NA" long term

Reply:
it depends on whether we allow subclasses to change na_value (or even consider allowing subclassing). strictly speaking it's probably `Any` so have left untyped for now.

Reply:
I don't think we need to consider subclasses right now, and keep it simple just for pandas (as long as there is no specific use case, our internal EAs are not meant to be subclassed) 

(unless fletcher would like to do this to add additional numba based algorithms to the ArrowStringArray)